### PR TITLE
Fix back-link for representers

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/controllers/mentoring/automation_controller.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ def with_feedback
   def edit
     @representation = Exercise::Representation.find_by!(uuid: params[:uuid])
     @examples = Exercise::Representation::FindExampleSubmissions.(@representation)
+    @automation_params = params.permit(:order, :criteria, :page, :track_slug, :only_mentored_solutions)
```
Comment (iHiD) [inline app/controllers/mentoring/automation_controller.rb]: Can we nest these down a level in some key pls. Maybe `source`?
```suggestion
    @automation_params = params.permit(source: {:order, :criteria, :page, :track_slug, :only_mentored_solutions})
```

(I’m not certain that syntax is correct)

File: app/helpers/react_components/mentoring/representations/representation.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ module ReactComponents
   module Mentoring
     module Representations
       class Representation < ReactComponent
-        initialize_with :mentor, :representation, :example_submissions
+        initialize_with :mentor, :representation, :example_submissions, :params
```
Comment (iHiD) [inline app/helpers/react_components/mentoring/representations/representation.rb]:
```suggestion
        initialize_with :mentor, :representation, :example_submissions, :source_params
```

Etc

File: app/controllers/mentoring/automation_controller.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@ def with_feedback
   def edit
     @representation = Exercise::Representation.find_by!(uuid: params[:uuid])
     @examples = Exercise::Representation::FindExampleSubmissions.(@representation)
-    @automation_params = params.permit(:order, :criteria, :page, :track_slug, :only_mentored_solutions)
+    @source_params = params.permit(source: [:order, :criteria, :page, :track_slug, :only_mentored_solutions])
```
Comment (iHiD) [inline app/controllers/mentoring/automation_controller.rb]: Maybe?
```suggestion
    @source_params = params.permit(source: [:order, :criteria, :page, :track_slug, :only_mentored_solutions])[:source]
```

File: app/helpers/react_components/mentoring/representations/representation.rb [hunk 4]
```diff
@@ -56,8 +56,8 @@ def mentor_data
 
         def back_link
           representation.feedback_type.nil? ?
-              Exercism::Routes.mentoring_automation_index_path(**params) :
-              Exercism::Routes.with_feedback_mentoring_automation_index_path(**params)
+              Exercism::Routes.mentoring_automation_index_path(**source_params.fetch(:source)) :
+              Exercism::Routes.with_feedback_mentoring_automation_index_path(**source_params.fetch(:source))
```
Comment (iHiD) [inline app/helpers/react_components/mentoring/representations/representation.rb]:
```suggestion
              Exercism::Routes.mentoring_automation_index_path(**source_params) :
              Exercism::Routes.with_feedback_mentoring_automation_index_path(**source_params)
```

File: app/controllers/mentoring/automation_controller.rb (hunk 3 above)
Comment (ErikSchierboom) [reply app/controllers/mentoring/automation_controller.rb]: I had already done that :)

**Review by dem4ron — APPROVED** [review]

**Review by iHiD — APPROVED** [review]

**Pull request merged**